syntax = "proto3";

package currencyconverter;

// Currency conversion service
service CurrencyConverter {
  // Converts an amount from USD to the specified target currency
  rpc ConvertCurrency (ConversionRequest) returns (ConversionResponse) {}
}

// Request message with amount and target currency
message ConversionRequest {
  double amount = 1;
  string to_currency = 2;
}

// Response with the converted amount
message ConversionResponse {
  double converted_amount = 1;
}